renderdoc/os/posix/linux/linux_stringio.cpp

```diff
@@ -605,11 +605,44 @@ rdcstr GetAppFolderFilename(const rdcstr &filename)
   return ret + filename;
 }
 
-void GetExecutableFilename(rdcstr &selfName)
+bool IsWineExecutable()
 {
   char path[512] = {0};
   readlink("/proc/self/exe", path, 511);
 
+  const char *mod = strrchr(path, '/');
+  if(mod != NULL)
+    mod++;
+  else if(*path)
+    mod = path;
+  else
+    mod = "unknown";
+
+  return !strcmp(mod, "wine-preloader") || !strcmp(mod, "wine64-preloader");
+}
+
+void GetExecutableFilename(rdcstr &selfName)
+{
+  char path[512] = {0};
+  // Read /proc/self/cmdline here instead of
+  // /proc/self/exe because otherwise when RenderDoc'ing
+  // under Wine, all applications show up as wine64-preloader.
+  //
+  // /proc/[blah]/cmdline has arguments separated by NUL.
+  FILE *file = fopen("/proc/self/cmdline", FileIO::ReadText);
+  if(file)
+  {
+    // Just read everything at once.
+    ::fread(path, 1, 511, file);
```

Comment:
`GetExecutableFilename` is expected to return a full path to the executable, if I understand it right reading the cmdline here arg 0 is only going to have the argument specified when running, which might not be a full path but could be `./foo` or just `foo`?

Can you add a call here to something that will resolve arg 0 into an absolute path (I guess it would need to be aware of the cwd and PATH to resolve it properly), and fall back to readlink on `/proc/self/exe` if it can't find an absolute path?

Reply:
I don't think we can do that because the return from `/proc/self/cmdline` for Wine apps uses Windows paths, so we'd always fall back here.

Perhaps we can special-case the Wine preloader here (using the `IsWineExecutable` above) and use `/proc/self/cmdline` instead?

Reply:
How is a linux library loaded into a wine process somewhere intended to find its current running executable?

Also taking a step back what exactly is the problem being solved here? Is there a bug related to what path is identified as the current executable?

Reply:
Proton exes all say wine64_preloader instead of showing human readable .exe names that correspond to the expectations of the developer and let them disambiguate properly.

Reply:
If that's the only reason to change this then I think the better fix would be to make this kind of change in the (poorly named) [`FileIO::GetDefaultFiles` function](https://github.com/baldurk/renderdoc/blob/v1.x/renderdoc/os/posix/posix_stringio.cpp#L190). The `target` reported there is what ends up being reported as the current application to the user through the remote control connection, but it can be any arbitrary string and doesn't have to be an actual path or actual executable. It only uses the executable name because in most cases that's the most sensible default.